cfme/tests/ansible/test_embedded_ansible_services.py

```diff
@@ -153,3 +162,25 @@ def test_service_ansible_playbook_confirm():
     assert view.retirement.hosts.value == "localhost"
     assert view.retirement.verbosity.selected_option == "0 (Normal)"
     assert view.retirement.remove_resources.selected_option == "Yes"
+
+
+@pytest.mark.tier(3)
+@pytest.mark.parametrize("host_type,order_value,result", SERVICE_CATALOG_VALUES, ids=[
+    value[0] for value in SERVICE_CATALOG_VALUES])
+def test_service_ansible_playbook_order(request, ansible_catalog_item, service_catalog, host_type,
+        order_value, result):
+    """Test ordering ansible playbook service against default host, blank field and
+    unavailable host.
+    """
+    view = navigate_to(service_catalog, "AnsibleOrder")
+    view.fill({"hosts": order_value})
+    view.submit_button.click()
+    view.flash.assert_success_message("Order Request was Submitted")
+    request_descr = "Provisioning Service [{0}] from [{0}]".format(ansible_catalog_item.name)
+    service_request = Request(request_descr)
+    service_request.wait_for_request()
+    request.addfinalizer(service_request.remove_request)
```

Comment:
@psav should we provide a more clean way to do such clean-ups,

while the request object is a very practical choice,
my personal opinion  is, that it shouldn't be part of a well expressed test body

Reply:
There is part of me that thinks that the items that we need to do this test could be fixtures and the cleanup happen there. Do you have any other suggestions on cleaning up?

Reply:
for this test in particular it is a bit tricky since most of the test body would move into fixtures

on the other hand structurally this would be nice as it limits the test body to receiving the configured/obtained objects and only asserting the content